software/bismark/align/main.nf

```diff
@@ -0,0 +1,65 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+def options    = initOptions(params.options)
+
+process BISMARK_ALIGN {
+    tag "$meta.id"
+    label 'process_high'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
+
+    conda (params.enable_conda ? "bioconda::bismark==0.23.0" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/bismark:0.23.0--0"
+    } else {
+        container "quay.io/biocontainers/bismark:0.23.0--0"
+    }
+
+    input:
+    tuple val(meta), path(reads)
+    tuple val(meta), path(index)
```

Comment:
The idea behind this this was to allow for mapping against multiple reference genomes. By keeping the `meta` samples can easily be joined with their respective genome index.

But the more I think about it, I would prefer something like this rather where the joining already happened outside of the module:

`val(meta), path(reads), path(index)`

What do you think?

Reply:
Yep, I agree that `val(meta), path(reads), path(index)` is a better option because you can just add the index to the `meta` map at some point and forget about it until you need to use it 👍🏽 Can just use a `map` to filter the elements.